Adjust Departure Generator

DavidMerzJr opened the pull request:
If a path has all poses in all raster strokes with x and y pointing in the same direction (not flipping back and forth each stroke), a departure with x/y components will 'backtrack' on some strokes and continue 'forward' on others.  This commit changes that so that 'positive' x/y values always move away from the penultimate pose in a stroke, and 'negative' x/y values always move back towards the penultimate pose in a stroke.

If the general method used is agreeable, I could add a second commit doing the same for the axial approach generator.

Levi-Armstrong commented on tesseract/tesseract_planning/tesseract_process_planners/include/tesseract_process_planners/generators/extension_departure_generator.h:
```diff
@@ -0,0 +1,108 @@
+#ifndef TESSERACT_PLANNING_EXTENSION_DEPARTURE_GENERATOR_H
```
Missing license.

DavidMerzJr replied:
Added.

Levi-Armstrong replied:
Thanks

Levi-Armstrong commented:
@DavidMerzJr I missunderstood your question; I would us the license format you used in the extension generator in all of the other files.

DavidMerzJr commented:
@Levi-Armstrong I updated all the license statements I added to include the Doxygen macros and my best guess at a short description.  I listed you as the author on them.
